js/handle_results.js

```diff
@@ -1,7 +1,26 @@
-const handleResults = (data) => {
+
+const unescapeMap = Object.freeze({
+  '&amp;' : '&',
+  '&lt;'  : '<',
+  '&gt;'  : '>',
+  '&quot;': '"',
+  '&#39;' : "'",
+  '&#x2F;': '/',
+  '&#x60;': '`',
+  '&#x3D;': '='
+});
+const unescapeHtml = (string) => String(string)
+    .replace(/&amp;|&lt;|&gt;|&quot;|&#39;|&#x2F;|&#x60;|&#x3D;/g, (s) => unescapeMap[s]);
+
+
+const handleResults = (data, query) => {
   const result = document.getElementById("result");
   result.innerHTML = "";
 
+  let a = document.getElementById("stackoverflow-query")
+  a.innerHTML = "[vespa] " + query;
+  a.href = encodeURI("https://stackoverflow.com/search?q=[vespa] " + unescapeHtml(query));
```

Comment:
This should use `URLSearchParam` to properly encode the query. `encodeURI()` does not escape special URI characters, e.g. `&`. Maybe something like
```
const searchParams = new URLSearchParams();
searchParams.set('s', '[vespa] ' + unescapeHtml(query));
a.href = 'https://stackoverflow.com/search?' + searchParams.toString();

Reply:
agree, that is better, fixing